### PR TITLE
Weekly Patch Release v1.3.3 [full merge, no squash]

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Next patch release after #7589

Related to  #7467

```
gh pr list -s merged -S 'merged:2021-05-19T16:30:00.000Z..2021-05-26T23:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "v1.3.x") or (.milestone.title == null)) | [.url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

ok #7692 8ba6304c73c0fd030c50d5de52cbaddfed161613 Increment the total batch idx before the accumulation early exit
ok #7683 2c10ecc232f5fb576638683e24744a5be8911c9a MAINTAINER has been deprecated
no #7676 ad168fc4c69fb586f38a2e7396d4d981ae52aaa0 chlog for 1.3.2 + legacy test
ok #7677 8b01497e4235139edb83cd0b8efd87380a1af726 Fix global step update when the epoch is skipped
ok #7415 3f460b150aab16d1fa267251d738bada6d890675 Move parameter validation specific to TPU Training plugins
ok #7674 a54bc5dba3133cf492205e6a60ca46e8ab22c465 Fix progress bar print error when called before training
ok #7566 0c958c5a1f8ca8926eeaf1a2035725c15417830e Fix dataloaders are not reset when tuning the model
ok #7563 01109cdf0c44a150c262b65e70a7e1e64003cf93 Fix/mismatched toggle optimizer
no #7639 03ea68f8a2a4a7fed1ff6637380a0301a4ac97a5 Removed hparams assignment example in doc
ok #7592 92cf396de2fe49e89a625a200d641bd8b6aeb328 Override `broadcast_object_list` for `torch<1.8`
ok #7608 ed271905cf07d86af5b6d8e088b6c04813d1a8a1 Clear predict_progress_bar in ProgressBar.__getstate__
no #7606 6e56f56aa18746a47fdfe697e88de66a8f43edb5 docker use $(nproc)
no #7598 922c0a607b7c2fe095ffe0f74795dba7b410f6ba Fix incorrect code-snippet in optimizers doc

Questions:

- ~Should https://github.com/PyTorchLightning/pytorch-lightning/pull/7415 be included since it is a follow up to a bugfix PR?~

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by ethanwharris — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,9 +34,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed setting correct `DistribType` for `ddp_cpu` (spawn) backend ([#7492](https://github.com/PyTorchLightning/pytorch-lightning/pull/7492))
 - Fixed incorrect number of calls to LR scheduler when `check_val_every_n_epoch > 1` ([#7032](https://github.com/PyTorchLightning/pytorch-lightning/pull/7032))
 
-
 ## [1.3.1] - 2021-05-11
 
+
```
Comment (carmocca) [inline CHANGELOG.md:39]: (not blocking) extra whitespace

```suggestion
```

**Review by carmocca — APPROVED** [review]
